RNTester/e2e/__tests__/Switch-test.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+/* global device, element, by, expect, waitFor */
+
+const jestExpect = require('expect');
+
+describe.only('Switch', () => {
+  beforeAll(async () => {
+    await element(by.id('explorer_search')).replaceText('<Switch>');
+    await element(by.label('<Switch> Native boolean input')).tap();
+  });
+
+  afterAll(async () => {
+    await element(by.label('Back')).tap();
+  });
+
+  it('Switch that starts on should switch', async () => {
+    const testID = 'on-off-initial-off';
+    const indicatorID = 'on-off-initial-off-indicator';
+
+    await expect(element(by.id(testID))).toHaveValue('0');
+    await expect(element(by.id(indicatorID))).toHaveText('Off');
+    await element(by.id(testID)).tap();
+    await expect(element(by.id(testID))).toHaveValue('1');
+    await expect(element(by.id(indicatorID))).toHaveText('On');
+  });
+
+  it('Switch that starts off should switch', async () => {
+    const testID = 'on-off-initial-on';
+    const indicatorID = 'on-off-initial-on-indicator';
+
+    await expect(element(by.id(testID))).toHaveValue('1');
+    await expect(element(by.id(indicatorID))).toHaveText('On');
+    await element(by.id(testID)).tap();
+    await expect(element(by.id(testID))).toHaveValue('0');
+    await expect(element(by.id(indicatorID))).toHaveText('Off');
+  });
+
+  it('disabled switch should not toggle', async () => {
+    const onTestID = 'disabled-initial-on';
+    const offTestID = 'disabled-initial-off';
+    const onIndicatorID = 'disabled-initial-on-indicator';
+    const offIndicatorID = 'disabled-initial-off-indicator';
+
+    await expect(element(by.id(onTestID))).toHaveValue('1');
+    await expect(element(by.id(onIndicatorID))).toHaveText('On');
+
+    try {
```

Comment:
This feels gross. When tapping on an element that is disabled, EarlGrey throws the following error:

```
err:  Cannot perform action due to constraint(s) failure.
    Exception with Action: {
      "Action Name":  "Tap",
      "Element Description":  "<RCTSwitch:0x7f91a1615b60; AX=Y; AX.id='disabled-initial-on'; AX.value='1'; AX.frame={{20.5, 295}, {51, 31}}; AX.activationPoint={46, 310.5}; AX.traits='UIAccessibilityTraitButton'; AX.focused='N'; frame={{0, 0}, {51, 31}}; opaque; alpha=1; disabled>",
      "Failed Constraint(s)":  "enabled",
      "All Constraint(s)":  "(!(isSystemAlertViewShown) && ((respondsToSelector(isAccessibilityElement) && isAccessibilityElement) || kindOfClass('UIView')) && (enabled && !(((kindOfClass('UIView') || respondsToSelector(accessibilityContainer)) && ancestorThatMatches(!(enabled))))) && interactable)",
      "Recovery Suggestion":  "Adjust element properties so that it matches the failed constraint(s)."
    }
    
    
    [
      {
        "Description":  "Cannot perform action due to constraint(s) failure.",
        "Error Domain":  "com.google.earlgrey.ElementInteractionErrorDomain",
        "Error Code":  "1",
        "File Name":  "GREYBaseAction.m",
        "Function Name":  "-[GREYBaseAction satisfiesConstraintsForElement:error:]",
        "Line":  "66"
      }
    ]
```

Reply:
We should definitely have better error logs, but what do you mean by cleaner way?

Reply:
One example of something cleaner would be using jest's expect `.toThrow`. In this example it would look like this:

```
await jestExpect(element(by.id(onTestID)).tap()).rejects.toThrow('Cannot perform action due to constraint(s) failure');
```

However, the error we get back is actually an Error object, with the message field set to an Error, with *that's* message field set to the long string, making it impossible to assert as if it was a normal error.